packages/amplication-client/src/Layout/MenuItem.tsx

```diff
@@ -1,29 +1,35 @@
 import React from "react";
 import { useRouteMatch, NavLink } from "react-router-dom";
 import classNames from "classnames";
+import { Button, EnumButtonStyle } from "../Components/Button";
 
 import { Icon } from "@rmwc/icon";
-import { SideNav } from "@primer/components";
+import "./MenuItem.scss";
 
 type Props = {
-  to: string;
+  to?: string;
   title: string;
   icon?: string;
   className?: string;
   children?: React.ReactNode;
   onClick?: () => void;
 };
 
+const NON_URL = "non-url";
+
 const MenuItem = ({ to, title, icon, className, children, onClick }: Props) => {
-  const match = useRouteMatch(to);
+  const match = useRouteMatch(to || NON_URL);
+
   return (
-    <SideNav.Link
+    <Button
+      buttonStyle={EnumButtonStyle.Clear}
+      as={to ? NavLink : Button}
       onClick={onClick}
-      as={NavLink}
```

Comment:
Break the component to one with the `to` prop and one without.

Reply:
I don't see many benefits from splitting this component to two.
We will have to repeat most of the code in both, unless we use two external ones that wrap the same internal component. In this case we will be left with the same internal code - Just two external interfaces. But, I don't think the current interfaces is confusing in any way...